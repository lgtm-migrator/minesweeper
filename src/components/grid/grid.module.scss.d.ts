// autogenerated by typings-for-css-modules-loader.
// Please do not change this file!
declare namespace GridModuleScssNamespace {
  export interface IGridModuleScss {
    containerWrapper: string;
  }
}

declare const GridModuleScssModule: GridModuleScssNamespace.IGridModuleScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: GridModuleScssNamespace.IGridModuleScss;
};

export = GridModuleScssModule;
