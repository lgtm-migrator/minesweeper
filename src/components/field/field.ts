import { resolve } from "path";

export type Cell = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
export type Field = Cell[][];
export type Coordinates = [number, number];

export const CellState: Record<string, Cell> = {
  empty: 0,
  bomb: 9,
  hidden: 10,
  mark: 11,
  weakMark: 12,
};

export const generateFieldWithDefaultState = (size: number, state: Cell = CellState.empty): Field =>
  new Array(size).fill(null).map(() => new Array(size).fill(state));

export const fieldGenerator = (size: number, probability: number): Field => {
  if (probability < 0 || probability > 1) {
    throw new Error("Probability must be between 0 and 1");
  }

  const result: Field = generateFieldWithDefaultState(size);

  // the size grid of the cells
  let unprocessedCells = size * size;
  // cells with bombs included
  let restCellsWithBombs = unprocessedCells * probability;

  for (let y = 0; y < size; y++) {
    for (let x = 0; x < size; x++) {
      if (restCellsWithBombs === 0) {
        return result;
      }
      if (restCellsWithBombs / unprocessedCells > 0) {
        result[y][x] = CellState.bomb;
        restCellsWithBombs--;
      }
      unprocessedCells--;
    }
  }

  return result;
};
