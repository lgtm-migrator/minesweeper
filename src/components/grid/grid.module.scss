.containerWrapper {
  display: grid;
  width: max-content;
  padding: 1vw;
}
