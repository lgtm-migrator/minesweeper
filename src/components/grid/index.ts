export * from "./cell/cell";
export * from "./grid";
