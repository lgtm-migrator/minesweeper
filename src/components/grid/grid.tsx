import { CellComponent } from "@components/grid/cell/cell";
import { Coordinates, Field } from "@helpers/field";
import React, { FunctionComponent, PropsWithChildren } from "react";

import styles from "./grid.module.scss";

export interface GridProps {
  /**
   * Field data
   */
  children: Field;
  /**
   * onClick handler
   */
  onClick: (coords: Coordinates) => void;
  /**
   * onContextMenu handler
   */
  onContextMenu: (coords: Coordinates) => void;
}

export const GridComponent: FunctionComponent<GridProps> = ({ children, ...restProps }) => {
  return (
    <WrapperContainer size={children.length} role="grid">
      {children.map((row, y) =>
        row.map((cell, x) => {
          return <CellComponent key={`${y}_${x}_${cell}`} children={cell} coords={[y, x]} {...restProps} />;
        })
      )}
    </WrapperContainer>
  );
};

const WrapperContainer: FunctionComponent<PropsWithChildren<{ size: number; role: string }>> = ({
  children,
  size,
  role,
}) => (
  <div
    className={styles.containerWrapper}
    style={{
      gridTemplateColumns: `repeat(${size}, auto)`,
    }}
    role={role}
  >
    {children}
  </div>
);
