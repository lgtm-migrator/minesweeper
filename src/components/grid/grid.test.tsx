import { GridComponent } from "@components/grid/grid";
import { Field } from "@helpers/field";
import { render } from "@testing-library/react";
import React from "react";

const MockFieldData: Field = [
  [9, 2, 9, 1, 0, 0, 1, 1, 1, 1, 1, 1],
  [1, 2, 2, 2, 1, 0, 1, 9, 1, 1, 9, 1],
  [0, 0, 1, 9, 10, 0, 2, 2, 2, 1, 1, 1],
  [0, 0, 10, 10, 1, 0, 1, 9, 1, 2, 2, 2],
  [0, 1, 1, 2, 2, 9, 1, 1, 1, 0, 0, 0],
  [0, 1, 9, 3, 9, 2, 10, 0, 0, 2, 1, 1],
  [0, 2, 2, 4, 9, 2, 10, 1, 1, 1, 9, 1],
  [0, 1, 9, 2, 1, 1, 1, 9, 1, 2, 2, 2],
  [0, 1, 10, 10, 0, 0, 1, 1, 1, 1, 9, 1],
  [0, 1, 10, 10, 0, 0, 1, 1, 1, 1, 9, 1],
  [0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 9, 1],
  [0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 9, 1],
];

describe("Grid Component rendering test", function () {
  it("should render the grid correctly", function () {
    const internalProps = {
      onClick: jest.fn(),
      onContextMenu: jest.fn(),
    };

    const { asFragment } = render(<GridComponent children={MockFieldData} {...internalProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
