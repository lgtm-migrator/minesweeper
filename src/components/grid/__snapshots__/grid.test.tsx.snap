// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component rendering test should render the grid correctly 1`] = `
<DocumentFragment>
  <div
    class="containerWrapper"
    role="grid"
    style="grid-template-columns: repeat(12, auto);"
  >
    <div
      aria-label="0,0"
      class="closedFrame revealedFrame bombFrame"
      data-testid="0,0"
      role="cell"
    >
      <div
        aria-label="0,0"
        class="bombEntity"
        data-testid="0,0"
      />
    </div>
    <div
      aria-label="0,1"
      class="closedFrame revealedFrame"
      data-testid="0,1"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="0,2"
      class="closedFrame revealedFrame bombFrame"
      data-testid="0,2"
      role="cell"
    >
      <div
        aria-label="0,2"
        class="bombEntity"
        data-testid="0,2"
      />
    </div>
    <div
      aria-label="0,3"
      class="closedFrame revealedFrame"
      data-testid="0,3"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="0,4"
      class="closedFrame revealedFrame"
      data-testid="0,4"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="0,5"
      class="closedFrame revealedFrame"
      data-testid="0,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="0,6"
      class="closedFrame revealedFrame"
      data-testid="0,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="0,7"
      class="closedFrame revealedFrame"
      data-testid="0,7"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="0,8"
      class="closedFrame revealedFrame"
      data-testid="0,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="0,9"
      class="closedFrame revealedFrame"
      data-testid="0,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="0,10"
      class="closedFrame revealedFrame"
      data-testid="0,10"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="0,11"
      class="closedFrame revealedFrame"
      data-testid="0,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="1,0"
      class="closedFrame revealedFrame"
      data-testid="1,0"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="1,1"
      class="closedFrame revealedFrame"
      data-testid="1,1"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="1,2"
      class="closedFrame revealedFrame"
      data-testid="1,2"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="1,3"
      class="closedFrame revealedFrame"
      data-testid="1,3"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="1,4"
      class="closedFrame revealedFrame"
      data-testid="1,4"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="1,5"
      class="closedFrame revealedFrame"
      data-testid="1,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="1,6"
      class="closedFrame revealedFrame"
      data-testid="1,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="1,7"
      class="closedFrame revealedFrame bombFrame"
      data-testid="1,7"
      role="cell"
    >
      <div
        aria-label="1,7"
        class="bombEntity"
        data-testid="1,7"
      />
    </div>
    <div
      aria-label="1,8"
      class="closedFrame revealedFrame"
      data-testid="1,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="1,9"
      class="closedFrame revealedFrame"
      data-testid="1,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="1,10"
      class="closedFrame revealedFrame bombFrame"
      data-testid="1,10"
      role="cell"
    >
      <div
        aria-label="1,10"
        class="bombEntity"
        data-testid="1,10"
      />
    </div>
    <div
      aria-label="1,11"
      class="closedFrame revealedFrame"
      data-testid="1,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="2,0"
      class="closedFrame revealedFrame"
      data-testid="2,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="2,1"
      class="closedFrame revealedFrame"
      data-testid="2,1"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="2,2"
      class="closedFrame revealedFrame"
      data-testid="2,2"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="2,3"
      class="closedFrame revealedFrame bombFrame"
      data-testid="2,3"
      role="cell"
    >
      <div
        aria-label="2,3"
        class="bombEntity"
        data-testid="2,3"
      />
    </div>
    <div
      aria-label="2,4"
      class="closedFrame changedBorderColor"
      data-testid="2,4"
      role="cell"
    />
    <div
      aria-label="2,5"
      class="closedFrame revealedFrame"
      data-testid="2,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="2,6"
      class="closedFrame revealedFrame"
      data-testid="2,6"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="2,7"
      class="closedFrame revealedFrame"
      data-testid="2,7"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="2,8"
      class="closedFrame revealedFrame"
      data-testid="2,8"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="2,9"
      class="closedFrame revealedFrame"
      data-testid="2,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="2,10"
      class="closedFrame revealedFrame"
      data-testid="2,10"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="2,11"
      class="closedFrame revealedFrame"
      data-testid="2,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="3,0"
      class="closedFrame revealedFrame"
      data-testid="3,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="3,1"
      class="closedFrame revealedFrame"
      data-testid="3,1"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="3,2"
      class="closedFrame changedBorderColor"
      data-testid="3,2"
      role="cell"
    />
    <div
      aria-label="3,3"
      class="closedFrame changedBorderColor"
      data-testid="3,3"
      role="cell"
    />
    <div
      aria-label="3,4"
      class="closedFrame revealedFrame"
      data-testid="3,4"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="3,5"
      class="closedFrame revealedFrame"
      data-testid="3,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="3,6"
      class="closedFrame revealedFrame"
      data-testid="3,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="3,7"
      class="closedFrame revealedFrame bombFrame"
      data-testid="3,7"
      role="cell"
    >
      <div
        aria-label="3,7"
        class="bombEntity"
        data-testid="3,7"
      />
    </div>
    <div
      aria-label="3,8"
      class="closedFrame revealedFrame"
      data-testid="3,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="3,9"
      class="closedFrame revealedFrame"
      data-testid="3,9"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="3,10"
      class="closedFrame revealedFrame"
      data-testid="3,10"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="3,11"
      class="closedFrame revealedFrame"
      data-testid="3,11"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="4,0"
      class="closedFrame revealedFrame"
      data-testid="4,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="4,1"
      class="closedFrame revealedFrame"
      data-testid="4,1"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="4,2"
      class="closedFrame revealedFrame"
      data-testid="4,2"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="4,3"
      class="closedFrame revealedFrame"
      data-testid="4,3"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="4,4"
      class="closedFrame revealedFrame"
      data-testid="4,4"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="4,5"
      class="closedFrame revealedFrame bombFrame"
      data-testid="4,5"
      role="cell"
    >
      <div
        aria-label="4,5"
        class="bombEntity"
        data-testid="4,5"
      />
    </div>
    <div
      aria-label="4,6"
      class="closedFrame revealedFrame"
      data-testid="4,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="4,7"
      class="closedFrame revealedFrame"
      data-testid="4,7"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="4,8"
      class="closedFrame revealedFrame"
      data-testid="4,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="4,9"
      class="closedFrame revealedFrame"
      data-testid="4,9"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="4,10"
      class="closedFrame revealedFrame"
      data-testid="4,10"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="4,11"
      class="closedFrame revealedFrame"
      data-testid="4,11"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="5,0"
      class="closedFrame revealedFrame"
      data-testid="5,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="5,1"
      class="closedFrame revealedFrame"
      data-testid="5,1"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="5,2"
      class="closedFrame revealedFrame bombFrame"
      data-testid="5,2"
      role="cell"
    >
      <div
        aria-label="5,2"
        class="bombEntity"
        data-testid="5,2"
      />
    </div>
    <div
      aria-label="5,3"
      class="closedFrame revealedFrame"
      data-testid="5,3"
      role="cell"
      style="color: rgb(236, 101, 97);"
    >
      3
    </div>
    <div
      aria-label="5,4"
      class="closedFrame revealedFrame bombFrame"
      data-testid="5,4"
      role="cell"
    >
      <div
        aria-label="5,4"
        class="bombEntity"
        data-testid="5,4"
      />
    </div>
    <div
      aria-label="5,5"
      class="closedFrame revealedFrame"
      data-testid="5,5"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="5,6"
      class="closedFrame changedBorderColor"
      data-testid="5,6"
      role="cell"
    />
    <div
      aria-label="5,7"
      class="closedFrame revealedFrame"
      data-testid="5,7"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="5,8"
      class="closedFrame revealedFrame"
      data-testid="5,8"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="5,9"
      class="closedFrame revealedFrame"
      data-testid="5,9"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="5,10"
      class="closedFrame revealedFrame"
      data-testid="5,10"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="5,11"
      class="closedFrame revealedFrame"
      data-testid="5,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="6,0"
      class="closedFrame revealedFrame"
      data-testid="6,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="6,1"
      class="closedFrame revealedFrame"
      data-testid="6,1"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="6,2"
      class="closedFrame revealedFrame"
      data-testid="6,2"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="6,3"
      class="closedFrame revealedFrame"
      data-testid="6,3"
      role="cell"
      style="color: rgb(35, 61, 183);"
    >
      4
    </div>
    <div
      aria-label="6,4"
      class="closedFrame revealedFrame bombFrame"
      data-testid="6,4"
      role="cell"
    >
      <div
        aria-label="6,4"
        class="bombEntity"
        data-testid="6,4"
      />
    </div>
    <div
      aria-label="6,5"
      class="closedFrame revealedFrame"
      data-testid="6,5"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="6,6"
      class="closedFrame changedBorderColor"
      data-testid="6,6"
      role="cell"
    />
    <div
      aria-label="6,7"
      class="closedFrame revealedFrame"
      data-testid="6,7"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="6,8"
      class="closedFrame revealedFrame"
      data-testid="6,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="6,9"
      class="closedFrame revealedFrame"
      data-testid="6,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="6,10"
      class="closedFrame revealedFrame bombFrame"
      data-testid="6,10"
      role="cell"
    >
      <div
        aria-label="6,10"
        class="bombEntity"
        data-testid="6,10"
      />
    </div>
    <div
      aria-label="6,11"
      class="closedFrame revealedFrame"
      data-testid="6,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="7,0"
      class="closedFrame revealedFrame"
      data-testid="7,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="7,1"
      class="closedFrame revealedFrame"
      data-testid="7,1"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="7,2"
      class="closedFrame revealedFrame bombFrame"
      data-testid="7,2"
      role="cell"
    >
      <div
        aria-label="7,2"
        class="bombEntity"
        data-testid="7,2"
      />
    </div>
    <div
      aria-label="7,3"
      class="closedFrame revealedFrame"
      data-testid="7,3"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="7,4"
      class="closedFrame revealedFrame"
      data-testid="7,4"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="7,5"
      class="closedFrame revealedFrame"
      data-testid="7,5"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="7,6"
      class="closedFrame revealedFrame"
      data-testid="7,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="7,7"
      class="closedFrame revealedFrame bombFrame"
      data-testid="7,7"
      role="cell"
    >
      <div
        aria-label="7,7"
        class="bombEntity"
        data-testid="7,7"
      />
    </div>
    <div
      aria-label="7,8"
      class="closedFrame revealedFrame"
      data-testid="7,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="7,9"
      class="closedFrame revealedFrame"
      data-testid="7,9"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="7,10"
      class="closedFrame revealedFrame"
      data-testid="7,10"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="7,11"
      class="closedFrame revealedFrame"
      data-testid="7,11"
      role="cell"
      style="color: rgb(43, 177, 61);"
    >
      2
    </div>
    <div
      aria-label="8,0"
      class="closedFrame revealedFrame"
      data-testid="8,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="8,1"
      class="closedFrame revealedFrame"
      data-testid="8,1"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="8,2"
      class="closedFrame changedBorderColor"
      data-testid="8,2"
      role="cell"
    />
    <div
      aria-label="8,3"
      class="closedFrame changedBorderColor"
      data-testid="8,3"
      role="cell"
    />
    <div
      aria-label="8,4"
      class="closedFrame revealedFrame"
      data-testid="8,4"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="8,5"
      class="closedFrame revealedFrame"
      data-testid="8,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="8,6"
      class="closedFrame revealedFrame"
      data-testid="8,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="8,7"
      class="closedFrame revealedFrame"
      data-testid="8,7"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="8,8"
      class="closedFrame revealedFrame"
      data-testid="8,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="8,9"
      class="closedFrame revealedFrame"
      data-testid="8,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="8,10"
      class="closedFrame revealedFrame bombFrame"
      data-testid="8,10"
      role="cell"
    >
      <div
        aria-label="8,10"
        class="bombEntity"
        data-testid="8,10"
      />
    </div>
    <div
      aria-label="8,11"
      class="closedFrame revealedFrame"
      data-testid="8,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="9,0"
      class="closedFrame revealedFrame"
      data-testid="9,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="9,1"
      class="closedFrame revealedFrame"
      data-testid="9,1"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="9,2"
      class="closedFrame changedBorderColor"
      data-testid="9,2"
      role="cell"
    />
    <div
      aria-label="9,3"
      class="closedFrame changedBorderColor"
      data-testid="9,3"
      role="cell"
    />
    <div
      aria-label="9,4"
      class="closedFrame revealedFrame"
      data-testid="9,4"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="9,5"
      class="closedFrame revealedFrame"
      data-testid="9,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="9,6"
      class="closedFrame revealedFrame"
      data-testid="9,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="9,7"
      class="closedFrame revealedFrame"
      data-testid="9,7"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="9,8"
      class="closedFrame revealedFrame"
      data-testid="9,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="9,9"
      class="closedFrame revealedFrame"
      data-testid="9,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="9,10"
      class="closedFrame revealedFrame bombFrame"
      data-testid="9,10"
      role="cell"
    >
      <div
        aria-label="9,10"
        class="bombEntity"
        data-testid="9,10"
      />
    </div>
    <div
      aria-label="9,11"
      class="closedFrame revealedFrame"
      data-testid="9,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,0"
      class="closedFrame revealedFrame"
      data-testid="10,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="10,1"
      class="closedFrame revealedFrame"
      data-testid="10,1"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,2"
      class="closedFrame revealedFrame"
      data-testid="10,2"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,3"
      class="closedFrame revealedFrame"
      data-testid="10,3"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,4"
      class="closedFrame revealedFrame"
      data-testid="10,4"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="10,5"
      class="closedFrame revealedFrame"
      data-testid="10,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="10,6"
      class="closedFrame revealedFrame"
      data-testid="10,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,7"
      class="closedFrame revealedFrame"
      data-testid="10,7"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,8"
      class="closedFrame revealedFrame"
      data-testid="10,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,9"
      class="closedFrame revealedFrame"
      data-testid="10,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="10,10"
      class="closedFrame revealedFrame bombFrame"
      data-testid="10,10"
      role="cell"
    >
      <div
        aria-label="10,10"
        class="bombEntity"
        data-testid="10,10"
      />
    </div>
    <div
      aria-label="10,11"
      class="closedFrame revealedFrame"
      data-testid="10,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,0"
      class="closedFrame revealedFrame"
      data-testid="11,0"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="11,1"
      class="closedFrame revealedFrame"
      data-testid="11,1"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,2"
      class="closedFrame revealedFrame"
      data-testid="11,2"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,3"
      class="closedFrame revealedFrame"
      data-testid="11,3"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,4"
      class="closedFrame revealedFrame"
      data-testid="11,4"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="11,5"
      class="closedFrame revealedFrame"
      data-testid="11,5"
      role="cell"
      style="color: rgba(0, 0, 0, 0);"
    >
      0
    </div>
    <div
      aria-label="11,6"
      class="closedFrame revealedFrame"
      data-testid="11,6"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,7"
      class="closedFrame revealedFrame"
      data-testid="11,7"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,8"
      class="closedFrame revealedFrame"
      data-testid="11,8"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,9"
      class="closedFrame revealedFrame"
      data-testid="11,9"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
    <div
      aria-label="11,10"
      class="closedFrame revealedFrame bombFrame"
      data-testid="11,10"
      role="cell"
    >
      <div
        aria-label="11,10"
        class="bombEntity"
        data-testid="11,10"
      />
    </div>
    <div
      aria-label="11,11"
      class="closedFrame revealedFrame"
      data-testid="11,11"
      role="cell"
      style="color: rgb(42, 72, 236);"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;
